## CI Failure: Lint check

[**Lint check**](https://github.com/NVIDIA-NeMo/RL/actions/runs/25643848705/job/75268853464) — two pre-commit hooks failed:

### 1. `ruff` — D205 (missing blank line in docstring)

```
infra/nrl_k8s/src/nrl_k8s/orchestrate.py:579:5: D205 1 blank line required between summary line and description
```

The `_rewrite_entrypoint_recipe` docstring has the summary line wrapping directly into the description body. Add a blank line after the summary:

```python
    """Rewrite ``--config <path>.yaml`` flags in the train entrypoint to
    point at the CLI's RECIPE.
                                          # <-- needs blank line here
    Without this rewrite, ...
```

Or make the summary a single line, then blank line, then description.

### 2. `ruff-format` — 2 files reformatted

Formatting changes needed in `orchestrate.py` and `test_orchestrate.py`:

- `orchestrate.py`: long `_rewrite_entrypoint_recipe(...)` call needs line-wrapping; `_CONFIG_FLAG_RE = re.compile(...)` should be a single line
- `test_orchestrate.py`: a few method signatures and string literals can fit on one line

<details>
<summary>Full formatting diff</summary>

```diff
--- a/infra/nrl_k8s/src/nrl_k8s/orchestrate.py
+++ b/infra/nrl_k8s/src/nrl_k8s/orchestrate.py
@@ -505,7 +505,9 @@
     # not touched here; they have their own configs and go through a
     # separate submission path.
-    entrypoint = _rewrite_entrypoint_recipe(launch.entrypoint, loaded, repo_root, upload=upload, log=log)
+    entrypoint = _rewrite_entrypoint_recipe(
+        launch.entrypoint, loaded, repo_root, upload=upload, log=log
+    )

@@ -538,9 +540,7 @@
-_CONFIG_FLAG_RE = re.compile(
-    r"(--config(?:\s+|=))(['\"]?)([^\s'\"]+\.ya?ml)\2"
-)
+_CONFIG_FLAG_RE = re.compile(r"(--config(?:\s+|=))(['\"]?)([^\s'\"]+\.ya?ml)\2")

--- a/infra/nrl_k8s/tests/unit/test_orchestrate.py
+++ b/infra/nrl_k8s/tests/unit/test_orchestrate.py
@@ -518,9 +518,7 @@
-    def test_rewrites_hardcoded_config_to_cli_recipe(
-        self, tmp_path: Path, log
-    ) -> None:
+    def test_rewrites_hardcoded_config_to_cli_recipe(self, tmp_path: Path, log) -> None:

@@ -585,10 +583,7 @@
-        ep = (
-            "python train.py --config-name=foo --config-dir=bar "
-            "--config-path conf"
-        )
+        ep = "python train.py --config-name=foo --config-dir=bar --config-path conf"
```

</details>

### Fix

@hemildesai — the quickest way to fix this is to set up pre-commit hooks locally so these get caught before push:

```bash
uv run --group dev pre-commit install
uv run --group dev pre-commit run --all-files
```

See the [Contributing Guide](https://github.com/NVIDIA-NeMo/RL/blob/main/CONTRIBUTING.md) for full setup instructions.

--- a/infra/nrl_k8s/src/nrl_k8s/orchestrate.py
+++ b/infra/nrl_k8s/src/nrl_k8s/orchestrate.py
@@ -505,7 +505,9 @@ def submit_training(
     # merging, never the pod. Daemon entrypoints (gym/generation) are
     # not touched here; they have their own configs and go through a
     # separate submission path.
-    entrypoint = _rewrite_entrypoint_recipe(launch.entrypoint, loaded, repo_root, upload=upload, log=log)
+    entrypoint = _rewrite_entrypoint_recipe(
+        launch.entrypoint, loaded, repo_root, upload=upload, log=log
+    )
 
     try:
         handle = submitter.submit(
@@ -538,9 +540,7 @@ def default_run_id(role: str) -> str:
 # can substitute it. The mandatory ``\s|=`` after ``--config`` keeps us
 # from accidentally matching Hydra's ``--config-name`` /
 # ``--config-dir`` / ``--config-path``.
-_CONFIG_FLAG_RE = re.compile(
-    r"(--config(?:\s+|=))(['\"]?)([^\s'\"]+\.ya?ml)\2"
-)
+_CONFIG_FLAG_RE = re.compile(r"(--config(?:\s+|=))(['\"]?)([^\s'\"]+\.ya?ml)\2")
 
 
 def _recipe_path_in_pod(
@@ -576,10 +576,10 @@ def _rewrite_entrypoint_recipe(
     upload: bool,
     log: callable,
 ) -> str:
-    """Rewrite ``--config <path>.yaml`` flags in the train entrypoint to
-    point at the CLI's RECIPE.
+    """Rewrite ``--config <path>.yaml`` flags in the train entrypoint.
 
-    Without this rewrite, ``nrl-k8s run RECIPE_A --infra X`` silently
+    Rewrites flags to point at the CLI's RECIPE. Without this rewrite,
+    ``nrl-k8s run RECIPE_A --infra X`` silently
     runs whatever recipe the entrypoint hardcoded — RECIPE_A would only
     affect client-side validation + Hydra override merging, never the
     pod. Substituting the path in-place makes the CLI argument

--- a/infra/nrl_k8s/tests/unit/test_orchestrate.py
+++ b/infra/nrl_k8s/tests/unit/test_orchestrate.py
@@ -518,9 +518,7 @@ class TestRewriteEntrypointRecipe:
         loaded.source_path = recipe
         return loaded
 
-    def test_rewrites_hardcoded_config_to_cli_recipe(
-        self, tmp_path: Path, log
-    ) -> None:
+    def test_rewrites_hardcoded_config_to_cli_recipe(self, tmp_path: Path, log) -> None:
         log_fn, lines = log
         repo = tmp_path / "repo"
         loaded = self._loaded_for(repo, "infra/recipes/x.yaml")
@@ -585,10 +583,7 @@ class TestRewriteEntrypointRecipe:
         log_fn, lines = log
         repo = tmp_path / "repo"
         loaded = self._loaded_for(repo, "x.yaml")
-        ep = (
-            "python train.py --config-name=foo --config-dir=bar "
-            "--config-path conf"
-        )
+        ep = "python train.py --config-name=foo --config-dir=bar --config-path conf"
 
         result = orchestrate._rewrite_entrypoint_recipe(
             ep, loaded, repo, upload=False, log=log_fn

